### PR TITLE
Replace Dictionary with IDictionary in ByKey method

### PR DESCRIPTION
**Pull request by KenitoInc**
This PR fixes issue #96 

We are allowing ByKey method to receive an IDictionary<string, object> instead of the concrete class Dictionary<string, object>

NOTE: _This fix is dependent on the update to OData Client as done in this PR_ https://github.com/OData/odata.net/pull/1648
UPDATE: The PR was merged and shipped in V7.3.0

### CODE REVIEW

**Review by paulodero — APPROVED** [review]
After moving the tests in code gen. It will be nice to check if any tests are broken. Can be done when synchronizing with code gen.

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs [hunk 1]
```diff
@@ -6808,8 +6808,8 @@ internal override void WriteCastToMethods(string baseTypeName, string derivedTyp
 this.Write(this.ToStringHelper.ToStringWithCulture(derivedTypeFullName));
 
 this.Write("\r\n        \'\'\' </summary>\r\n        \'\'\' <param name=\"source\">source entity</param>\r" +
-        "\n        <Global.System.Runtime.CompilerServices.Extension()>\r\n        Public " +
-        "Function CastTo");
+        "\n        <Global.System.Runtime.CompilerServices.Extension()>\r\n        Public Fu" +
```
Comment (marabooy) [inline ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: consider splitting the string on a whitespace or new line than in the middle of a word

**Review by marabooy — CHANGES_REQUESTED** [review]

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs [hunk 2]
```diff
@@ -237,7 +237,7 @@ public string MetadataDocumentUri
         {
             value = uri.Scheme + "://" + uri.Authority + uri.AbsolutePath;
             value = value.TrimEnd('/');
-            if (!value.EndsWith("$metadata", StringComparison.CurrentCulture))
+            if (!value.EndsWith("$metadata"))
```
Comment (g2mula) [inline ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: Maybe this change was not meant for checkin?

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs [hunk 3]
```diff
@@ -2455,14 +2455,14 @@ public void WriteLine(string textToAppend)
     /// </summary>
     public void Write(string format, params object[] args)
     {
-        this.Write(string.Format(global::System.Globalization.CultureInfo.CurrentCulture, format, args));
+        this.Write(string.Format(CultureInfo.InvariantCulture, format, args));
```
Comment (g2mula) [inline ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: Maybe this change was not meant for checkin?

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs [hunk 4]
```diff
@@ -2455,14 +2455,14 @@ public void WriteLine(string textToAppend)
...
     }
     /// <summary>
     /// Write formatted text directly into the generated output
     /// </summary>
     public void WriteLine(string format, params object[] args)
     {
-        this.WriteLine(string.Format(global::System.Globalization.CultureInfo.CurrentCulture, format, args));
+        this.WriteLine(string.Format(CultureInfo.InvariantCulture, format, args));
```
Comment (g2mula) [inline ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: Maybe this change was not meant for checkin?

**Review by g2mula — CHANGES_REQUESTED** [review]
There are changes made to do with CurrentCulture/InvariantCulture... These do not look to have been meant for checkin... 
They should probably be reverted... The rest look good to me.

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs (hunk 2 above)
Comment (KenitoInc) [reply ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: This change was from a previous commit on the tt file but the generated cs file wasn't committed

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs (hunk 3 above)
Comment (KenitoInc) [reply ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: This change was from a previous commit on the tt file but the generated cs file wasn't committed

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs (hunk 4 above)
Comment (KenitoInc) [reply ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: This change was from a previous commit on the tt file but the generated cs file wasn't committed

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs (hunk 1 above)
Comment (KenitoInc) [reply ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: This is autogenerated code

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs (hunk 2 above)
Comment (g2mula) [reply ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: If it ain't for this particular PR (Replace Dictionary with IDictionary in ByKey method) then it should probably be in a different PR...

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs (hunk 3 above)
Comment (g2mula) [reply ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: Then the commit changes shouldn't probably be part of this commit for this PR but it's own commit for that other PR...

File: ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs (hunk 4 above)
Comment (g2mula) [reply ODataConnectedService/src/Templates/ODataT4CodeGenerator.cs]: Then the changes should be part of that other PR, not this one.

**Review by g2mula — DISMISSED** [review]
This looks good, it's just that it contains some changes which are not related to the issue #96 being fixed.
Those other changes should be in other PRs...

**Review by g2mula — APPROVED** [review]
Approved.

**Pull request closed without merging**